### DISCUSSION TITLE
Who generates ID?

### DISCUSSION THREAD

**Discussion by pgkmop** [rustdesk/rustdesk#5989] [Q&A] [answered]
Hello!
I want to use my own server. So I've installed it and configured both sides to communicate via it. When I run such client, it gets 8-digit ID. I thought if server is mine, ID would start from 1 or something like it.
So my question here: does RustDesk connects to some server of its to generate or to get an ID? So even if I use my own server, the program still connect to something on the internets?

**Comment by SergeyMy** [comment, marked as answer]
ID is generated either randomly or :) this is your part of the MAC address :):):):):) //see source code
